retail/interactive-tutorial/RetailProducts.Samples/SetInventorySample.cs

```diff
@@ -0,0 +1,139 @@
+    }
+
+    /// <summary>
+    /// Get the set inventory request.
+    /// </summary>
+    /// <param name="productName">The actual product with inventory info.</param>
+    /// <returns>Set inventory request.</returns>
+    private static SetInventoryRequest GetSetInventoryRequest(string productName)
+    {
+        // The request timestamp
+        DateTime requestTimeStamp = DateTime.UtcNow;
+
+        // The out-of-order request timestamp
+        // requestTimeStamp = DateTime.UtcNow.AddDays(-1);
+
+        var setMask = new FieldMask
+        {
+            Paths = { "price_info", "availability", "fulfillment_info", "available_quantity" }
+        };
+
+        var setInventoryRequest = new SetInventoryRequest
+        {
+            Inventory = GetProductWithInventoryInfo(productName),
+            SetTime = Timestamp.FromDateTime(requestTimeStamp),
+            AllowMissing = true,
+            SetMask = setMask
+        };
+
+        Console.WriteLine("Set inventory request:");
+        Console.WriteLine($"Product Name: {setInventoryRequest.Inventory.Name}");
+        Console.WriteLine($"Product Title: {setInventoryRequest.Inventory.Title}");
+        Console.WriteLine($"Product Brands: {setInventoryRequest.Inventory.Brands}");
+        Console.WriteLine($"Product Categories: {setInventoryRequest.Inventory.Categories}");
+        Console.WriteLine($"Product Fulfillment Info: {setInventoryRequest.Inventory.FulfillmentInfo}");
+        Console.WriteLine($"Product Price Info: {setInventoryRequest.Inventory.PriceInfo}");
+        Console.WriteLine();
+
+        return setInventoryRequest;
+    }
+
+    /// <summary>
+    /// Call the Retail API to set product inventory.
+    /// </summary>
+    /// <param name="product">The actual product.</param>
+    public static void SetProductInventory(string productName)
+    {
+        var setInventoryRequest = GetSetInventoryRequest(productName);
+        var client = ProductServiceClient.Create();
+        var operation = client.SetInventory(setInventoryRequest);
+
+        Console.WriteLine("Set inventory. Please, wait.");
+        Console.WriteLine();
+
+        operation.PollUntilCompleted();
+    }
+}
+// [END retail_set_inventory]
+
+/// <summary>
+/// The set inventory tutorial class.
+/// </summary>
+public static class SetInventoryTutorial
+{
+    [Runner.Attributes.Example]
+    public static void PerformSetInventoryOperation()
+    {
+        string projectId = Environment.GetEnvironmentVariable("GOOGLE_PROJECT_ID");
+
+        // Create product.
+        Product createdProduct = CreateProductSample.CreateRetailProductWithFulfillment(projectId);
+
+        // Set inventory for product.
+        SetInventorySample.SetProductInventory(createdProduct.Name);
+
+        // Get product.
+        Product product = GetProductSample.GetRetailProduct(createdProduct.Name);
+
+        Console.WriteLine($"Product Price Info: {product.PriceInfo}");
+        Console.WriteLine();
```

Comment:
Maybe for being consistent then, don't print anything here?

Reply:
This information in console output for understanding what we changed, because `GetRetailProduct()` in this case gives insufficient information about the product. Will it be better to update console output directly in `GetProductSample.GetRetailProduct()`?

Reply:
I think it's fine then to leave it as is.